test/fixtures/gcs-fixture/src/test/java/fixture/gcs/GoogleCloudStorageHttpHandlerTests.java

```diff
@@ -0,0 +1,509 @@
+                .status()
+        );
+    }
+
+    public void testCheckEndpoint() {
+        final var handler = new GoogleCloudStorageHttpHandler("bucket");
+
+        assertEquals(
+            RestStatus.OK,
+            handleRequest(handler, "GET", "/", BytesArray.EMPTY, Headers.of("Metadata-Flavor", "Google")).restStatus()
+        );
+    }
+
+    public void testSimpleObjectOperations() {
+        final var bucket = randomAlphaOfLength(10);
+        final var handler = new GoogleCloudStorageHttpHandler(bucket);
+
+        assertEquals(RestStatus.NOT_FOUND, handleRequest(handler, "GET", "/download/storage/v1/b/" + bucket + "/o/blob").restStatus());
+
+        assertEquals(
+            new TestHttpResponse(RestStatus.OK, "{\"kind\":\"storage#objects\",\"items\":[],\"prefixes\":[]}"),
+            handleRequest(handler, "GET", "/storage/v1/b/" + bucket + "/o")
+        );
+
+        // Multipart upload
+        final var body = randomAlphaOfLength(50);
+        assertEquals(
+            RestStatus.OK,
+            handleRequest(
+                handler,
+                "POST",
+                "/upload/storage/v1/b/" + bucket + "/?uploadType=multipart",
+                createGzipCompressedMultipartUploadBody(bucket, "path/blob", body)
+            ).restStatus()
+        );
+        assertEquals(
+            new TestHttpResponse(RestStatus.OK, body),
+            handleRequest(handler, "GET", "/download/storage/v1/b/" + bucket + "/o/path/blob")
+        );
+
+        assertEquals(new TestHttpResponse(RestStatus.OK, Strings.format("""
+            {"kind":"storage#objects","items":[{"kind":"storage#object","bucket":"%s","name":"path/blob","id":"path/blob","size":"50"}
+            ],"prefixes":[]}""", bucket)), handleRequest(handler, "GET", "/storage/v1/b/" + bucket + "/o"));
+
+        assertEquals(new TestHttpResponse(RestStatus.OK, Strings.format("""
+            {"kind":"storage#objects","items":[{"kind":"storage#object","bucket":"%s","name":"path/blob","id":"path/blob","size":"50"}
+            ],"prefixes":[]}""", bucket)), handleRequest(handler, "GET", "/storage/v1/b/" + bucket + "/o?prefix=path/"));
+
+        assertEquals(
+            new TestHttpResponse(RestStatus.OK, """
+                {"kind":"storage#objects","items":[],"prefixes":[]}"""),
+            handleRequest(handler, "GET", "/storage/v1/b/" + bucket + "/o?prefix=path/other")
+        );
+
+        assertEquals(
+            new TestHttpResponse(RestStatus.OK, """
+                --__END_OF_PART__d8b50acb-87dc-4630-a3d3-17d187132ebc__
+                Content-Length: 162
+                Content-Type: application/http
+                content-id: 1
+                content-transfer-encoding: binary
+
+                HTTP/1.1 204 NO_CONTENT
+
+
+
+
+                --__END_OF_PART__d8b50acb-87dc-4630-a3d3-17d187132ebc__
+                """.replaceAll("\n", "\r\n")),
```

Comment:
Does this do the right thing on Windows? I've triggered a Windows CI run to find out.